Python/Product/PythonTools/PythonTools/Editor/PythonEditorServices.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.ComponentModel.Composition;
+using Microsoft.PythonTools.Intellisense;
+using Microsoft.VisualStudio.ComponentModelHost;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Classification;
+using Microsoft.VisualStudio.Utilities;
+using Microsoft.VisualStudioTools;
+
+namespace Microsoft.PythonTools.Editor {
+    /// <summary>
+    /// Provides centralized access to services used by the editor.
+    /// </summary>
+    /// <remarks>
+    /// MEF services provided by VS should be imported eagerly using the
+    /// [Import] attribute on a public field.
```

Comment:
If you're going to document a pattern here [ImportingConstructor] is a little bit nicer in that it lets you keep fields readonly